pages/marketplace/search/[query].tsx

```diff
@@ -1,38 +1,46 @@
 'use client';
 
-import CategoriesNav from '@modules/marketplace/component/CategoriesNav/CategoriesNav';
-import MainLayout from '../../../components/Layout/MainLayout';
 import { useEffect, useState } from 'react';
 import ProductCard from '@modules/marketplace/component/ProductCard';
 import styles from '../../../modules/marketplace/component/landingpage/productCardWrapper/product-card-wrapper.module.css';
 import { ProductResult } from '../../../@types';
 import Link from 'next/link';
 import Error from '@modules/marketplace/component/landingpageerror/ErrorPage';
 import CategoryLayout from '@modules/marketplace/component/layout/category-layout';
+import { useRouter } from 'next/router';
+import { searchPosts } from '../../../http/api/searchPost';
 
 export default function Index() {
   const [results, setResults] = useState<ProductResult[]>([]);
-  const searchquery = typeof window !== 'undefined' ? localStorage.getItem('keyword') : null;
+  const router = useRouter();
+  const { query } = router.query;
+  const searchQuery = Array.isArray(query) ? query[0] : query;
 
   useEffect(() => {
-    if (typeof window !== 'undefined') {
-      const getresult = localStorage.getItem('search_result');
-      if (getresult) {
-        const result = JSON.parse(getresult);
-        setResults(result);
-      }
+    if (searchQuery) {
+      const fetchData = async () => {
+        try {
+          const results = await searchPosts(searchQuery);
+          setResults(results);
+          console.log(results)
+        } catch (error) {
+          console.error(error);
+        }
+      };
+
+      fetchData();
     }
-  }, [results]);
+  }, [searchQuery]);
 
   return (
     <>
-      {results?.length === 0 ? (
+    {results?.length === 0 ? (
         <Error />
```

Comment:
length 0 does not mean you should throw an error

show that the query returned nothing instead and another case to check if there was an error